testing/typescript/data/stores.ts

```diff
@@ -20,6 +21,17 @@ export async function infersTItemFromComplexLoadResult() {
   const value: { a: number } = result;
 }
 
+export async function loadReturnRightTypeIfLoadDefinedInOptions() {
+  const store = new CustomStore({
+    // eslint-disable-next-line @typescript-eslint/require-await
+    load: async () => ({
+      data: [{ a: 1 }],
+    }),
+  });
+
+  const value: ResolvedData<{ a: number }> = await store.load();
```

Comment:
Do not use explicit type specification, it seems that it checks nothing in this case.

I think we should check  the item type here, smth like this:
```suggestion
  const loadResult = await store.load();

  const actualItem: (typeof loadResult) extends ResolvedData<infer Item> ? Item : never = ANY;

  notAny(actualItem);
  notNever(actualItem);
  assertType<{a: number}>(toAssertion(actualItem));
```